src/electionguard_cli/e2e_steps/key_ceremony_step.py

```diff
@@ -0,0 +1,95 @@
+from typing import List
+from cli_models.e2e_inputs import E2eInputs
+from e2e_steps.e2e_step_base import E2eStepBase
+from electionguard.key_ceremony import (
+    CeremonyDetails,
+    ElectionJointKey,
+    ElectionPublicKey,
+)
+from electionguard.guardian import Guardian
+from electionguard.key_ceremony_mediator import KeyCeremonyMediator
+from electionguard.utils import get_optional
+
+
+class KeyCeremonyStep(E2eStepBase):
+    """Responsible for running a key ceremony and producing an elgamal public key given a list of guardians."""
+
+    def __announce_guardians(
```

Comment:
Is the double underscore `__` a click convention? This is usually used in certain methods like `__init__`, but I haven't seen it used like this?

https://docs.python.org/3/tutorial/classes.html#tut-private

Reply:
This could also be dramatically simplified by using `electionguard_tools` and probably should be.

Reply:
Correct, the "dunder" is used to indicate a private method since afaik Python doesn't have language support for private methods: https://www.geeksforgeeks.org/private-methods-in-python/

Reply:
This isn't accurate. See this: https://www.youtube.com/watch?v=HTLu2DFOdTg&t=1988s

This doesn't make this actually private, but is actually localizing the function. The single `_` underscore might be closer to what you mean.

Reply:
Here is the matching post for the same source you're using:
https://www.geeksforgeeks.org/underscore-_-python/

Reply:
✅ dramatically simplified